buildSrc/subprojects/plugins/src/main/kotlin/org/gradle/plugins/performance/BuildForkPointDistribution.kt

```diff
@@ -0,0 +1,101 @@
+
+    @OutputDirectory
+    val forkPointDistributionHome = project.objects.directoryProperty()
+
+    @OutputFile
+    val forkPointToolingApiJar = project.objects.fileProperty()
+
+    init {
+        forkPointDistributionHome.set(forkPointDistributionVersion.map({ project.rootProject.layout.projectDirectory.dir("build/distributions/gradle-$it") }))
+        forkPointToolingApiJar.set(forkPointDistributionVersion.map({ project.rootProject.layout.projectDirectory.file("build/distributions/gradle-tooling-api-$it.jar") }))
+    }
+
+    @TaskAction
+    fun buildDistribution() {
+        prepareGradleRepository()
+        tryBuildDistribution()
+        println("Building fork point succeeded, now the baseline is ${forkPointDistributionVersion.get()}")
+    }
+
+    private
+    fun prepareGradleRepository() {
+        val cloneDir = getGradleCloneTmpDir()
+        if (!File(cloneDir, ".git").isDirectory) {
+            FileUtils.forceMkdir(cloneDir)
+            project.execAndGetStdout(cloneDir.parentFile, "git", "clone", project.rootDir.absolutePath, getGradleCloneTmpDir().absolutePath, "--no-checkout")
+        }
+        project.execAndGetStdout(cloneDir, "git", "reset", "--hard")
+        project.execAndGetStdout(cloneDir, "git", "fetch")
+    }
+
+    private
+    fun getGradleCloneTmpDir() = File(project.rootProject.buildDir, "tmp/gradle-find-forkpoint")
+
+    private
+    fun tryBuildDistribution() {
+        val version = forkPointDistributionVersion.get()
+        val commit = version.substring(version.lastIndexOf('-') + 1)
+        project.execAndGetStdout(getGradleCloneTmpDir(), "git", "checkout", commit, "--force")
+        project.execAndGetStdout(getGradleCloneTmpDir(), *getBuildCommands())
+    }
+
+    private
+    fun getBuildCommands(): Array<String> {
+        project.delete(forkPointDistributionHome.get().asFile)
+        val buildCommands = arrayOf(
+            "./gradlew",
+            "--init-script",
+            File(getGradleCloneTmpDir(), "gradle/init-scripts/build-scan.init.gradle.kts").absolutePath,
+            "clean",
+            ":install",
+            "-Pgradle_installPath=" + forkPointDistributionHome.get().asFile.absolutePath,
+            ":toolingApi:installToolingApiShadedJar",
+            "-PtoolingApiShadedJarInstallPath=" + forkPointToolingApiJar.get().asFile.absolutePath)
+
+        val buildCacheParams = if (project.gradle.startParameter.isBuildCacheEnabled) {
+            val buildCacheConf = (project.gradle as GradleInternal).settings.buildCache
+            val remoteCache = buildCacheConf.remote as HttpBuildCache?
+            arrayOf("--build-cache",
+                "-Dgradle.cache.remote.url=" + if (remoteCache == null) "" else remoteCache.url,
+                "-Dgradle.cache.remote.username=" + if (remoteCache == null) "" else remoteCache.credentials.username,
+                "-Dgradle.cache.remote.password=" + if (remoteCache == null) "" else remoteCache.credentials.password)
```

Comment:
Could be simplified to e.g. `"-Dgradle.cache.remote.url=${remoteCache?.url ?: ""}",`